com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/protocol/Events.java

```diff
@@ -154,10 +172,24 @@ public static OutputEvent createStdoutOutput(String output) {
             return new OutputEvent(Category.stdout, output);
         }
 
+        /**
+         * Constructor an stdout output event with source info.
+         */
+        public static OutputEvent createStdoutOutput(String output, Source source, int line) {
+            return new OutputEvent(Category.stdout, output, source, line);
+        }
+
         public static OutputEvent createStderrOutput(String output) {
             return new OutputEvent(Category.stderr, output);
         }
 
+        /**
+         * Constructor an stderr output event with source info.
+         */
+        public static OutputEvent createStderrOutput(String output, Source source, int line) {
```

Comment:
createStderrOutput is occupied in above line, use a new createStderrOutputXXX for more details to help user to select the right functions between createStderrOutput and this method.